The job **`dist-sparcv9-solaris`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17514077871/job/49749396266) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/49749396266) [(plain)](https://github.com/rust-lang/rust/commit/28261ec8887a0f69b566e212666a01cdd80c49c2/checks/49749396266/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
     |
  96 | pub struct File(FileDesc);
     | -------------------------- similarly named tuple struct `File` defined here
...
1493 |         cvt(unsafe { libc::fcntl(file.as_raw_fd(), cmd, &flock) })
     |                                  ^^^^
     |                                  |
     |                                  not a value
     |                                  help: a tuple struct with a similar name exists (notice the capitalization): `File`

error[E0107]: enum takes 2 generic arguments but 1 generic argument was supplied
    --> library/std/src/sys/fs/unix.rs:1489:74
     |
1489 |     fn emulate_flock(&self, lock_type: libc::c_int, cmd: libc::c_int) -> Result<()> {
     |                                                                          ^^^^^^ -- supplied 1 generic argument
     |                                                                          |
     |                                                                          expected 2 generic arguments
     |
note: enum defined here, with 2 generic parameters: `T`, `E`
    --> /rustc/28261ec8887a0f69b566e212666a01cdd80c49c2/library/core/src/result.rs:549:10
     |
 549 | pub enum Result<T, E> {
     |          ^^^^^^ -  -
help: add missing generic argument
---

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1298:28
     |
1298 |         self.emulate_flock(libc::F_WRLCK, libc::F_SETLK);
     |              ------------- ^^^^^^^^^^^^^ expected `i32`, found `i16`
     |              |
     |              arguments to this method are incorrect
     |
note: method defined here
    --> library/std/src/sys/fs/unix.rs:1489:8
     |
1489 |     fn emulate_flock(&self, lock_type: libc::c_int, cmd: libc::c_int) -> Result<()> {
     |        ^^^^^^^^^^^^^        ----------------------
help: you can convert an `i16` to an `i32`
     |
1298 |         self.emulate_flock(libc::F_WRLCK.into(), libc::F_SETLK);
     |                                         +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1297:27
     |
---

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1331:28
     |
1331 |         self.emulate_flock(libc::F_RDLCK, libc::F_SETLK);
     |              ------------- ^^^^^^^^^^^^^ expected `i32`, found `i16`
     |              |
     |              arguments to this method are incorrect
     |
note: method defined here
    --> library/std/src/sys/fs/unix.rs:1489:8
     |
1489 |     fn emulate_flock(&self, lock_type: libc::c_int, cmd: libc::c_int) -> Result<()> {
     |        ^^^^^^^^^^^^^        ----------------------
help: you can convert an `i16` to an `i32`
     |
1331 |         self.emulate_flock(libc::F_RDLCK.into(), libc::F_SETLK);
     |                                         +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1330:34
     |
---

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1372:41
     |
1372 |         let result = self.emulate_flock(libc::F_WRLCK, libc::F_SETLKW);
     |                           ------------- ^^^^^^^^^^^^^ expected `i32`, found `i16`
     |                           |
     |                           arguments to this method are incorrect
     |
note: method defined here
    --> library/std/src/sys/fs/unix.rs:1489:8
     |
1489 |     fn emulate_flock(&self, lock_type: libc::c_int, cmd: libc::c_int) -> Result<()> {
     |        ^^^^^^^^^^^^^        ----------------------
help: you can convert an `i16` to an `i32`
     |
1372 |         let result = self.emulate_flock(libc::F_WRLCK.into(), libc::F_SETLKW);
     |                                                      +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1375:21
     |
---
     |                     this argument influences the type of `Err`
note: tuple variant defined here
    --> /rustc/28261ec8887a0f69b566e212666a01cdd80c49c2/library/core/src/result.rs:558:5
     |
 558 |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
     |     ^^^
help: call `Into::into` on this expression to convert `fs::TryLockError` into `io::error::Error`
     |
1375 |                 Err(TryLockError::WouldBlock.into())
     |                                             +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1377:21
     |
---
     |                     this argument influences the type of `Err`
note: tuple variant defined here
    --> /rustc/28261ec8887a0f69b566e212666a01cdd80c49c2/library/core/src/result.rs:558:5
     |
 558 |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
     |     ^^^
help: call `Into::into` on this expression to convert `fs::TryLockError` into `io::error::Error`
     |
1377 |                 Err(TryLockError::Error(err).into())
     |                                             +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1425:41
     |
1425 |         let result = self.emulate_flock(libc::F_RDLCK, libc::F_SETLKW);
     |                           ------------- ^^^^^^^^^^^^^ expected `i32`, found `i16`
     |                           |
     |                           arguments to this method are incorrect
     |
note: method defined here
    --> library/std/src/sys/fs/unix.rs:1489:8
     |
1489 |     fn emulate_flock(&self, lock_type: libc::c_int, cmd: libc::c_int) -> Result<()> {
     |        ^^^^^^^^^^^^^        ----------------------
help: you can convert an `i16` to an `i32`
     |
1425 |         let result = self.emulate_flock(libc::F_RDLCK.into(), libc::F_SETLKW);
     |                                                      +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1428:21
     |
---
     |                     this argument influences the type of `Err`
note: tuple variant defined here
    --> /rustc/28261ec8887a0f69b566e212666a01cdd80c49c2/library/core/src/result.rs:558:5
     |
 558 |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
     |     ^^^
help: call `Into::into` on this expression to convert `fs::TryLockError` into `io::error::Error`
     |
1428 |                 Err(TryLockError::WouldBlock.into())
     |                                             +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1430:21
     |
---
     |                     this argument influences the type of `Err`
note: tuple variant defined here
    --> /rustc/28261ec8887a0f69b566e212666a01cdd80c49c2/library/core/src/result.rs:558:5
     |
 558 |     Err(#[stable(feature = "rust1", since = "1.0.0")] E),
     |     ^^^
help: call `Into::into` on this expression to convert `fs::TryLockError` into `io::error::Error`
     |
1430 |                 Err(TryLockError::Error(err).into())
     |                                             +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1470:28
     |
1470 |         self.emulate_flock(libc::F_UNLCK, libc::F_SETLK)
     |              ------------- ^^^^^^^^^^^^^ expected `i32`, found `i16`
     |              |
     |              arguments to this method are incorrect
     |
note: method defined here
    --> library/std/src/sys/fs/unix.rs:1489:8
     |
1489 |     fn emulate_flock(&self, lock_type: libc::c_int, cmd: libc::c_int) -> Result<()> {
     |        ^^^^^^^^^^^^^        ----------------------
help: you can convert an `i16` to an `i32`
     |
1470 |         self.emulate_flock(libc::F_UNLCK.into(), libc::F_SETLK)
     |                                         +++++++

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1491:26
     |
---

error[E0308]: mismatched types
    --> library/std/src/sys/fs/unix.rs:1493:22
     |
1493 |         cvt(unsafe { libc::fcntl(file.as_raw_fd(), cmd, &flock) })
     |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- help: consider using a semicolon here: `;`
     |                      |
     |                      expected `()`, found `i32`

error[E0277]: the trait bound `(): sys::pal::unix::IsMinusOne` is not satisfied
    --> library/std/src/sys/fs/unix.rs:1493:13
     |
1493 |         cvt(unsafe { libc::fcntl(file.as_raw_fd(), cmd, &flock) })
     |         --- ^^^^^^^^^------------------------------------------^^
     |         |   |        |
     |         |   |        this tail expression is of type `i32`
     |         |   the trait `sys::pal::unix::IsMinusOne` is not implemented for `()`
     |         required by a bound introduced by this call
---
               isize
note: required by a bound in `sys::pal::unix::cvt`
    --> library/std/src/sys/pal/unix/mod.rs:308:15
     |
 308 | pub fn cvt<T: IsMinusOne>(t: T) -> crate::io::Result<T> {
     |               ^^^^^^^^^^ required by this bound in `cvt`

Some errors have detailed explanations: E0107, E0277, E0308, E0423.
For more information about an error, try `rustc --explain E0107`.
[RUSTC-TIMING] std test:false 3.464
error: could not compile `std` (lib) due to 19 previous errors
```

</details>

--- a/library/std/src/fs/tests.rs
+++ b/library/std/src/fs/tests.rs
@@ -226,6 +226,7 @@ fn file_test_io_seek_and_write() {
     target_os = "freebsd",
     target_os = "linux",
     target_os = "netbsd",
+    target_os = "solaris",
     target_vendor = "apple",
 ))]
 fn file_lock_multiple_shared() {
@@ -249,6 +250,7 @@ fn file_lock_multiple_shared() {
     target_os = "freebsd",
     target_os = "linux",
     target_os = "netbsd",
+    target_os = "solaris",
     target_vendor = "apple",
 ))]
 fn file_lock_blocking() {
@@ -273,6 +275,7 @@ fn file_lock_blocking() {
     target_os = "freebsd",
     target_os = "linux",
     target_os = "netbsd",
+    target_os = "solaris",
     target_vendor = "apple",
 ))]
 fn file_lock_drop() {
@@ -294,6 +297,7 @@ fn file_lock_drop() {
     target_os = "freebsd",
     target_os = "linux",
     target_os = "netbsd",
+    target_os = "solaris",
     target_vendor = "apple",
 ))]
 fn file_lock_dup() {

--- a/library/std/src/sys/fs/unix.rs
+++ b/library/std/src/sys/fs/unix.rs
@@ -1293,6 +1293,15 @@ impl File {
         return Ok(());
     }
 
+    #[cfg(target_os = "solaris")]
+    pub fn lock(&self) -> io::Result<()> {
+        let mut flock: libc::flock = unsafe { mem::zeroed() };
+        flock.l_type = libc::F_WRLCK as libc::c_short;
+        flock.l_whence = libc::SEEK_SET as libc::c_short;
+        cvt(unsafe { libc::fcntl(self.as_raw_fd(), libc::F_SETLKW, &flock) })?;
+        Ok(())
+    }
+
     #[cfg(not(any(
         target_os = "freebsd",
         target_os = "fuchsia",
@@ -1300,6 +1309,7 @@ impl File {
         target_os = "netbsd",
         target_os = "openbsd",
         target_os = "cygwin",
+        target_os = "solaris",
         target_vendor = "apple",
     )))]
     pub fn lock(&self) -> io::Result<()> {
@@ -1320,6 +1330,15 @@ impl File {
         return Ok(());
     }
 
+    #[cfg(target_os = "solaris")]
+    pub fn lock_shared(&self) -> io::Result<()> {
+        let mut flock: libc::flock = unsafe { mem::zeroed() };
+        flock.l_type = libc::F_RDLCK as libc::c_short;
+        flock.l_whence = libc::SEEK_SET as libc::c_short;
+        cvt(unsafe { libc::fcntl(self.as_raw_fd(), libc::F_SETLKW, &flock) })?;
+        Ok(())
+    }
+
     #[cfg(not(any(
         target_os = "freebsd",
         target_os = "fuchsia",
@@ -1327,6 +1346,7 @@ impl File {
         target_os = "netbsd",
         target_os = "openbsd",
         target_os = "cygwin",
+        target_os = "solaris",
         target_vendor = "apple",
     )))]
     pub fn lock_shared(&self) -> io::Result<()> {
@@ -1355,6 +1375,23 @@ impl File {
         }
     }
 
+    #[cfg(target_os = "solaris")]
+    pub fn try_lock(&self) -> Result<(), TryLockError> {
+        let mut flock: libc::flock = unsafe { mem::zeroed() };
+        flock.l_type = libc::F_WRLCK as libc::c_short;
+        flock.l_whence = libc::SEEK_SET as libc::c_short;
+        let result = cvt(unsafe { libc::fcntl(self.as_raw_fd(), libc::F_SETLK, &flock) });
+        if let Err(err) = result {
+            if err.kind() == io::ErrorKind::WouldBlock {
+                Err(TryLockError::WouldBlock)
+            } else {
+                Err(TryLockError::Error(err))
+            }
+        } else {
+            Ok(())
+        }
+    }
+
     #[cfg(not(any(
         target_os = "freebsd",
         target_os = "fuchsia",
@@ -1362,6 +1399,7 @@ impl File {
         target_os = "netbsd",
         target_os = "openbsd",
         target_os = "cygwin",
+        target_os = "solaris",
         target_vendor = "apple",
     )))]
     pub fn try_lock(&self) -> Result<(), TryLockError> {
@@ -1393,6 +1431,23 @@ impl File {
         }
     }
 
+    #[cfg(target_os = "solaris")]
+    pub fn try_lock_shared(&self) -> Result<(), TryLockError> {
+        let mut flock: libc::flock = unsafe { mem::zeroed() };
+        flock.l_type = libc::F_RDLCK as libc::c_short;
+        flock.l_whence = libc::SEEK_SET as libc::c_short;
+        let result = cvt(unsafe { libc::fcntl(self.as_raw_fd(), libc::F_SETLK, &flock) });
+        if let Err(err) = result {
+            if err.kind() == io::ErrorKind::WouldBlock {
+                Err(TryLockError::WouldBlock)
+            } else {
+                Err(TryLockError::Error(err))
+            }
+        } else {
+            Ok(())
+        }
+    }
+
     #[cfg(not(any(
         target_os = "freebsd",
         target_os = "fuchsia",
@@ -1400,6 +1455,7 @@ impl File {
         target_os = "netbsd",
         target_os = "openbsd",
         target_os = "cygwin",
+        target_os = "solaris",
         target_vendor = "apple",
     )))]
     pub fn try_lock_shared(&self) -> Result<(), TryLockError> {
@@ -1423,6 +1479,15 @@ impl File {
         return Ok(());
     }
 
+    #[cfg(target_os = "solaris")]
+    pub fn unlock(&self) -> io::Result<()> {
+        let mut flock: libc::flock = unsafe { mem::zeroed() };
+        flock.l_type = libc::F_UNLCK as libc::c_short;
+        flock.l_whence = libc::SEEK_SET as libc::c_short;
+        cvt(unsafe { libc::fcntl(self.as_raw_fd(), libc::F_SETLKW, &flock) })?;
+        Ok(())
+    }
+
     #[cfg(not(any(
         target_os = "freebsd",
         target_os = "fuchsia",
@@ -1430,6 +1495,7 @@ impl File {
         target_os = "netbsd",
         target_os = "openbsd",
         target_os = "cygwin",
+        target_os = "solaris",
         target_vendor = "apple",
     )))]
     pub fn unlock(&self) -> io::Result<()> {

